share/functions/__fish_svn_prompt.fish

```diff
@@ -0,0 +1,169 @@
+
+set -g __fish_svn_prompt_char_missing_display '!'
+set -g __fish_svn_prompt_char_missing_color yellow
+
+set -g __fish_svn_prompt_char_versioned_obstructed_display '~'
+set -g __fish_svn_prompt_char_versioned_obstructed_color magenta
+
+set -g __fish_svn_prompt_char_locked_display 'L'
+set -g __fish_svn_prompt_char_locked_color --bold red
+
+set -g __fish_svn_prompt_char_scheduled_display '+'
+set -g __fish_svn_prompt_char_scheduled_color --bold green
+
+set -g __fish_svn_prompt_char_switched_display 'S'
+set -g __fish_svn_prompt_char_switched_color --bold blue
+
+set -g __fish_svn_prompt_char_token_present_display 'K'
+set -g __fish_svn_prompt_char_token_present_color --bold cyan
+
+set -g __fish_svn_prompt_char_token_other_display 'O'
+set -g __fish_svn_prompt_char_token_other_color --underline purple
+
+set -g __fish_svn_prompt_char_token_stolen_display 'T'
+set -g __fish_svn_prompt_char_token_stolen_color --bold purple
+
+set -g __fish_svn_prompt_char_token_broken_display 'B'
+set -g __fish_svn_prompt_char_token_broken_color --bold magenta
+
+
+# ==============================
+
+# this sets up an array of all the types of status codes that could be returned.
+set -g __fish_svn_prompt_flag_names added conflicted deleted ignored modified replaced unversioned_external unversioned missing locked scheduled switched token_present token_other token_stolen token_broken
+
+function __fish_svn_prompt_parse_status --argument flag_status_string --description "helper function that does pretty formatting on svn status"
+	# iterate over the different status types
+	for flag_index in (seq (count $__fish_svn_prompt_flag_names))
+		# resolve the name of the variable for the character representing the current status type
+		set -l flag_name __fish_svn_prompt_char_{$__fish_svn_prompt_flag_names[$flag_index]}_char
+		# check to see if the status string for this column contains the character representing the current status type
+		if test (echo $flag_status_string | grep -c $$flag_name) -eq 1
+			# if it does, then get the names of the variables for the display character and colour to format it with
+			set -l flag_display __fish_svn_prompt_char_{$__fish_svn_prompt_flag_names[$flag_index]}_display
+			set -l flag_color __fish_svn_prompt_char_{$__fish_svn_prompt_flag_names[$flag_index]}_color
+			# set the colour and print display character, then restore to default display colour
+			printf '%s%s%s' (set_color $$flag_color) $$flag_display (set_color normal)
+		end
+	end
+end
+
+
+function __fish_svn_prompt --description "Prompt function for svn"
+	# if svn isn't installed then don't do anything
+	if not command -s svn > /dev/null
+		return 1
+	end
+
+	# make sure that this is a svn repo
+	set -l checkout_info (command svn info ^/dev/null)
+	if [ $status -ne 0 ];
+		return
+	end
+
+	# get the current revision number
+	set -l repo_revision_number (command svn info | grep "Last Changed Rev: " | sed "s=Last Changed Rev: ==")
```

Comment:
>I did notice there seems to be a function in fish shell for fetching the revision number

__fish_print_svn_rev? That has the same issue as it does basically the same thing.

>As for the grep strings, I was basing this off the source of the svn command printing out what appear to be english strings.

They are english strings if your locale is english. Mine is german, so it'll print german strings (e.g. "Letzte geänderte Rev").

I'm trying to find a portable way to get svn information (i.e. the svn counterpart to `git rev-parse` or `hg status`), but I'm not coming up with much. `svnversion` might fit the bill, but I'm not sure if it prints everything you want here, `svnlook` is another, but it works only with repositories, not working copies.